### PR TITLE
AGDIGGER-262 Android debug builds fail with build tools 26

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by odra** [comment]
@matskiv The keystore soft link was causing the issue, moving the file to this image fixes it, so I think we have the following options:

- Create the keystore when building the image (as done in this PR)
- Create a keystore file and add it into the repo and copy it to the container image
- Change the container's entrypoint file (https://github.com/aerogear/digger-android-slave-image/blob/master/scripts/run-jnlp.sh) to copy the keystore file from the android-sdk-linux folder to the jenkins home folder before running the command

I think the third option might make the container usage more flexible.

File: Dockerfile [hunk 1]
```diff
@@ -2,6 +2,17 @@ FROM openshift/jenkins-slave-base-centos7
...
+#build args
+ARG KEYSTORE_ALIAS=AndroidDebugKey
+ARG FULL_NAME=AG
+ARG ORG_UNIT=AeroGear
+ARG ORG_NAME=RedHat
+ARG LOCALITY=WaterFord
+ARG PROVINCE=WD
+ARG COUNTRY_CODE=IRL
+ARG STORE_PASS=android
+ARG KEY_PASS=android
```
Comment (matskiv) [inline Dockerfile:14]: I am not sure if we should make these arguments: STORE_PASS, KEY_PASS and KEYSTORE_ALIAS
AFAIK, it won't work correctly with non-default values

**Comment by matskiv** [comment]
It's good that you found out that the problem is with a symlink, this issue looked pretty difficult.


@odra  With the third option we wouldn't have to change the installer, and key details (like organization name, etc.) would be provided during installation. That would be better than having the key "hardcoded" into the image.
If we choose that option, we won't need this PR, but I'll test it now anyway :)

**Comment by matskiv** [comment]
docker build failed on my machine :/
```
Building native extensions.  This could take a while...
ERROR:  Error installing fastlane:
	jwt requires Ruby version >= 2.1.
...
The command '/bin/sh -c curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.2/install.sh | bash &&     [ -s "$NVM_DIR/nvm.sh" ] && . "$NVM_DIR/nvm.sh" &&     echo 'export NVM_DIR=${NVM_DIR}' >> ${HOME}/.bashrc &&     echo '[ -s "$NVM_DIR/nvm.sh" ] && . "$NVM_DIR/nvm.sh"' >> ${HOME}/.bashrc &&     echo 'CI=Y' >> ${HOME}/.bashrc &&     nvm install ${NODEJS_DEFAULT_VERSION} &&     chmod -R 777 ${NVM_DIR} &&     npm install -g cordova@${CORDOVA_DEFAULT_VERSION} &&     gem install fastlane -v ${FASTLANE_DEFAULT_VERSION} &&     npm install -g grunt@${GRUNT_DEFAULT_VERSION}'
returned a non-zero code: 1
```

**Comment by odra** [comment]
can you comment the fastlane installation line for now? I've sent another pr into this branch that should fix that (#8)

**Comment by matskiv** [comment]
I commented out fastlane installation, now it fails here:

```
/bin/sh: keytool: command not found
The command '/bin/sh -c keytool -genkey -noprompt -alias $KEYSTORE_ALIAS     -dname "CN=$FULL_NAME, OU=$ORG_UNIT, O=$ORG_NAME, L=$LOCALITY, S=$PROVINCE, C=$COUNTRY_CODE"     -keystore $HOME/.android/debug.keystore     -storepass $STORE_PASS     -keypass $KEY_PASS     -keysize 2048     -keyalg RSA' 
returned a non-zero code: 127
```

**Comment by odra** [comment]
Better to use PR #8 then, it just fixes both issues of this one

**Comment by matskiv** [comment]
PR #8 builds fine :+1:

File: Dockerfile (hunk 1 above)
Comment (ghost) [reply Dockerfile:14]: You can pass them as build args when you want to build container with user defined keystore

**Comment by ghost** [comment]
Close in favor of #9

**Pull request closed without merging**